src/ui/public/filter_bar/filter_bar.js

```diff
@@ -137,6 +128,20 @@ define(function (require) {
           }
         });
 
+        function convertToEditableFilter(filter) {
+          var model = _.cloneDeep(filter);
+
+          var filterType;
+          //Hide private properties
+          _.each(model, function (value, key) {
```

Comment:
This isn't a big deal at all, but if we lean on lodash, I think this whole function `convertToEditableFilter` can be rewritten to something like:

```js
function convertToEditableFilter(filter) {
  const model = _.cloneDeep(filter);
  const isPrivateField = key => key.match(privateFilterFieldRegex);
  return _.omit(model, (val, key) => isPrivateField(key));
}
```

Then we wouldn't need to write our own imperative walking/deleting of properties in the object.